flows/actions/modifiers/field.go

```diff
@@ -40,13 +41,17 @@ func (m *FieldModifier) Apply(env envs.Environment, assets flows.SessionAssets,
 	oldValue := contact.Fields().Get(m.field)
 
 	if !m.value.Equals(oldValue) {
-		// truncate text value if necessary
-		if m.value != nil && m.value.Text.Length() > env.MaxValueLength() {
-			m.value.Text = m.value.Text.Slice(0, env.MaxValueLength())
+		var value *flows.Value
+
+		// copy and truncate text value if necessary
+		if m.value != nil {
+			v := *m.value
+			value = &v
```

Comment:
need to make a copy otherwise the modifier itself is modified

Reply:
Ooh, this seems kinda subtle. Almost makes me wonder whether `Get` should be returning a value vs a pointer then this wouldn't be a mistake that could be made.

Reply:
We're using `value: null` for clearing fields right now

The good news is that it's fairly easy to spot in our tests because the modifier get's re-serialized

Reply:
Just noting it. Empty value would work just as well for the empty case. Just feels a little too subtle but we can track it.